playbooks/install_monitoring.yml

```diff
@@ -0,0 +1,30 @@
+---
+- name: Set up monitoring server
+  hosts: monitoring_server
+  become: true
+  gather_facts: true
+  pre_tasks:
+    - name: Generate list of docker clients from `monitoring_client` group
+      ansible.builtin.set_fact:
+        docker_client_hostnames: >
+          {{
+          item |
+          map('extract', hostvars, monitoring_server_hostname_extractor)
+          }}
+      loop:
+        - "{{ groups[monitoring_server_client_group] | default([]) }}"
+      failed_when: docker_client_hostnames | length == 0
```

Comment:
does the handle the case where we use the `--limit` argument to `ansible-playbooks`, or will it always include all clients?

Reply:
Thanks for flagging this. I've made a change to intersect `ansible_limit` with the `monitoring_client` group so that the playbook can be run with or without the `--limit` flag.